src/libcrun/status.c

```diff
@@ -71,22 +96,62 @@ get_state_directory_status_file (const char *state_root, const char *id)
   return ret;
 }
 
+static int
+read_pid_stat (pid_t pid, struct pid_stat *st, libcrun_error_t *err)
+{
+  int ret;
+  FILE *f;
+  cleanup_free char *pid_stat_file = NULL;
+
+  ret = xasprintf (&pid_stat_file, "/proc/%d/stat", pid);
+  if (UNLIKELY (ret == -1))
+    return crun_make_error (err, errno, "xasprintf failed");
+
+  f = fopen (pid_stat_file, "r");
+  if (f == NULL)
+    return crun_make_error (err, errno, "open state file %s", pid_stat_file);
+
+  ret = fscanf (f,"%d %255s %c %d %d %d %d %d %u %lu %lu %lu %lu %lu %lu %ld %ld %ld %ld %ld %ld %llu",
+    &(st->pid), st->comm, &(st->state), &(st->ppid), &(st->pgrp), &(st->session),
+    &(st->tty_nr), &(st->tpgid), &(st->flags), &(st->minflt), &(st->cminflt),
+    &(st->majflt), &(st->cmajflt), &(st->utime), &(st->stime), &(st->cutime),
+    &(st->cstime), &(st->priority), &(st->nice), &(st->num_threads), &(st->itrealvalue),
+    &(st->starttime));
+  if (UNLIKELY (ret != 22))
+    {
+      fclose(f);
```

Comment:
can we move the `fclose (f);` just after the `fscanf` so we don't have to worry about it in two places?

Alternatively we could use: `cleanup_file FILE *f = NULL;` and we don't have to worry about the `fclose` at all

Reply:
thanks!